libcnb-test/src/build.rs

```diff
@@ -1,62 +1,138 @@
+    let workspace_root_path = find_cargo_workspace_root_dir(&cargo_manifest_dir)
+        .map_err(PackageTestBuildpackError::FindCargoWorkspace)?;
 
-    let buildpack_binaries = build_buildpack_binaries(
-        &cargo_manifest_dir,
-        &cargo_metadata,
+    let package_dir =
+        tempdir().map_err(PackageTestBuildpackError::CannotCreateBuildpackTempDirectory)?;
+
+    let buildpack_dir_resolver = create_packaged_buildpack_dir_resolver(
+        package_dir.as_ref(),
         cargo_profile,
-        &cargo_env,
         target_triple.as_ref(),
-    )
-    .map_err(PackageCrateBuildpackError::BuildBinariesError)?;
+    );
+
+    // TODO: this could accidentally detect a packaged meta-buildpack twice. how should we ignore directories
+    //       containing packaged buildpacks that might get detected now that the user controls --package-dir?
+    //       - support .gitignore or some other persistent configuration?
+    //       - during packaging should we create some type of file that indicates the output is a packaged buildpack?
+    let buildpack_dependency_graph =
+        build_libcnb_buildpacks_dependency_graph(&workspace_root_path, &[])
+            .map_err(PackageTestBuildpackError::CreateBuildpackDependencyGraph)?;
 
-    assemble_buildpack_directory(
-        buildpack_dir.path(),
-        cargo_manifest_dir.join("buildpack.toml"),
-        &buildpack_binaries,
+    let root_node = buildpack_dependency_graph
+        .node_weights()
+        .find(|node| node.buildpack_id == buildpack_id.clone());
+
+    assert!(
+        root_node.is_some(),
+        "Could not package directory as buildpack! No buildpack with id `{buildpack_id}` exists in the workspace at {}",
+        workspace_root_path.display()
+    );
+
+    let build_order = get_dependencies(
+        &buildpack_dependency_graph,
+        &[root_node.expect("The root node should exist")],
     )
-    .map_err(PackageCrateBuildpackError::CannotAssembleBuildpackDirectory)?;
+    .map_err(PackageTestBuildpackError::GetDependencies)?;
+
+    let mut packaged_buildpack_dirs = BTreeMap::new();
+    for node in &build_order {
+        let buildpack_destination_dir = buildpack_dir_resolver(&node.buildpack_id);
+
+        fs::create_dir_all(&buildpack_destination_dir).unwrap();
+
+        libcnb_package::package::package_buildpack(
+            &node.path,
+            cargo_profile,
+            target_triple.as_ref(),
+            &cargo_build_env,
+            &buildpack_destination_dir,
+            &packaged_buildpack_dirs,
+        )
+        .map_err(PackageTestBuildpackError::PackageBuildpack)?;
+
+        packaged_buildpack_dirs.insert(node.buildpack_id.clone(), buildpack_destination_dir);
+    }
+
+    Ok(PackagedBuildpackDir {
+        _root: package_dir,
+        path: buildpack_dir_resolver(buildpack_id),
+    })
+}
 
-    Ok(buildpack_dir)
+pub(crate) struct PackagedBuildpackDir {
+    _root: TempDir,
+    pub(crate) path: PathBuf,
```

Comment:
It would be good to not have to have this additional struct, I removed most of these grouping structs from `libcnb-package` because they're confusing. We can do so easily by not creating a temporary directory in `package_buildpack` but instead pass in a target directory explicitly. The responsibility to create the temporary directory would automatically "bubble up" and end up in the place where the context of a single test lives.

All in all, we gain:
- A better isolated `package_buildpack` that does less implicit things.
  - Which also makes it easier to test (if we want to) or use in other contexts.
- Removal of the struct and, more importantly, the congnitive burden of having to remember what `PackagedBuildpackDir` actually means. When I saw the name for the first time, I immediately jumped to "it can't be just a path, otherwise if would be a `PathBuf`". But the name doesn't give a hint what that extra bit is.


Note: This will also effect `package_crate_buildpack`.

Reply:
👍 Good point.  The only reason I introduced this wrapper struct was so that the temporary directory wouldn't immediately go out of scope which would cause the buildpacks to be cleaned up before the actual test executes.  But it's not needed if I just move the temp dir creation to the runner.